.github/workflows/continuous-integration.yml

```diff
@@ -93,17 +82,38 @@ jobs:
           echo $HDF5_DIR
           echo "HDF5_DIR=$HDF5_DIR" >> $GITHUB_ENV
 
-      - name: Download a specific release of PETSc
+      - name: Cache PETSc
+        uses: actions/cache@v4
+        id: cache-petsc
+        env:
+          cache-name: cache-PETSc
+        with:
+          path: "./petsc"
+          key: cache-${{ matrix.os }}-${{ matrix.python-version }}
+
+      - if: steps.cache-petsc.outputs.cache-hit != 'true'
+        name: Download a specific release of PETSc
         run: |
           git clone --depth 1 --branch v3.20.5 https://gitlab.com/petsc/petsc.git
 
-      - name: Install PETSc with complex support, and test it
+      - if: steps.cache-petsc.outputs.cache-hit != 'true'
+        name: Install PETSc with complex support
         working-directory: ./petsc
         run: |
           export PETSC_DIR=$(pwd)
           export PETSC_ARCH=petsc-cmplx
           ./configure --with-scalar-type=complex --with-fortran-bindings=0 --have-numpy=1         
-          make all check
+          make all
+          echo "PETSC_DIR=$PETSC_DIR" > petsc.env
+          echo "PETSC_ARCH=$PETSC_ARCH" >> petsc.env
```

Comment:
Do you use a single `>` in order to create a new `petsc.env` file?

Reply:
yes @yguclu